src/main/java/refactoringml/RunImportCsv.java

```diff
@@ -0,0 +1,165 @@
+		cleanOldTempDir();
+		runner.run(argv, "/tmp/repos");
+	}
+
+	private static class Args {
+
+		@Parameter(names = { "-h", "--db-host" }, description = "Host IP or address of database")
+		private String databaseHost = "127.0.0.1";
+
+		@Parameter(names = { "-P", "--db-port" }, description = "Port of database")
+		private Short databasePort;
+
+		@Parameter(names = { "-u", "--db-user" }, description = "Database user")
+		private String databaseUser = System.getProperty("user.name");
+
+		@Parameter(names = { "-p", "--db-password" }, description = "Database password", password = true)
+		private String databasePassword = "pass";
+
+		@Parameter(names = { "-n", "--db-name" }, description = "Name of the database to use")
+		private String databaseName = "refactoringdb";
+
+		// TODO use converter into Enum directly instead of valueOf later on based on
+		// string to construct enum.
+		@Parameter(names = { "-d",
+				"--db-engine" }, description = "The database engine being used, currently supported values are MYSQL, MARIADB or POSTGRESQL")
+		private String databaseEngine = "MARIADB";
+
+		// TODO use converter into Enum directly instead of valueOf later on based on
+		// string to construct enum.
+		@Parameter(names = { "-c",
+				"--connection-pool" }, description = "The connection pooling library used. currently supported values are MYSQL, MARIADB or POSTGRESQL")
+		private String connectionPoolType = "HIKARICP";
+
+		@Parameter(names = { "-b", "--hbm2ddl" }, description = "The hibernate.hbm2ddl.auto in the database connection")
+		private String hbm2ddlStrategy = "create";
+
+		@Parameter(names = { "-s", "--store-source-code" }, description = "Store the source code of the metrics")
+		private boolean storeSourceCode = false;
+
+		@Parameter(names = { "-t",
+				"thread-amount" }, description = "Amount of threads to run in parallel. Will default to amount of cores available")
+		private int runWithCores = Runtime.getRuntime().availableProcessors();
+
+		@Parameter(required = true, description = "Input csv file")
+		private String inputCsvFile;
+
+	}
+
+	private void run(Args args, String repoLocation) throws InterruptedException, IOException {
+
+		ThreadPoolExecutor tp = (ThreadPoolExecutor) Executors.newFixedThreadPool(args.runWithCores);
+		tp.setCorePoolSize(args.runWithCores);
+		SessionFactory sf = HibernateConfig.getSessionFactory(args.databaseHost, Optional.ofNullable(args.databasePort),
+				args.databaseName, args.databaseUser, args.databasePassword, args.hbm2ddlStrategy,
+				DatabaseEngine.valueOf(args.databaseEngine), ConnectionPoolType.valueOf(args.connectionPoolType));
+		try {
+			try (Stream<String> stream = Files.lines(Paths.get(args.inputCsvFile))) {
+				List<RepoProcesser> tasks = stream.map(line -> new RepoProcesser(repoLocation, args.storeSourceCode,
+						new Database(sf.openSession()), line)).collect(Collectors.toList());
+				tp.invokeAll(tasks, 100, TimeUnit.HOURS);
```

Comment:
What happens here in case of an Error, e.g. OutOfMemoryError?

Reply:
I have not encountered it as of yet. This class was made with the idea for collecting data on smaller datasets without having to start a lot of infrastructure.